lms/templates/dashboard.html

```diff
@@ -49,6 +53,13 @@
         DashboardSearchFactory();
     </%static:require_module>
   % endif
+  % if redirect_message:
+    <%static:require_module module_name="js/views/message_banner" class_name="MessageBannerView">
```

Comment:
message_banner is not listed in build.js, so it will not get optimized (or cached). You can see this behavior on your sandbox.